packages/nimble/R/genCpp_sizeProcessing.R

```diff
@@ -991,6 +1000,23 @@ sizeNimbleListReturningFunction <- function(code, symTab, typeEnv) {
   if(length(asserts) == 0) NULL else asserts
 }
 
+sizeOptim <- function(code, symTab, typeEnv) {
+    # TODO input an optional nimbleList for control
+    asserts <- nimble:::recurseSetSizes(code, symTab, typeEnv)
+    code$type <- 'nimbleList'
+    code$sizeExprs <- symTab$getSymbolObject('OptimResultNimbleList', inherits = TRUE)
+    code$toEigenize <- 'maybe'
+    code$nDim <- 0
+    asserts <- c(asserts, sizeInsertIntermediate(code$caller, code$callerArgID, symTab, typeEnv))
```

Comment:
Usually the situation is if we have a function like `nimOptim` that we can code-generate only for a direct assignment, e.g. `result <- nimOptim(...)`.  In that case `code$caller$name %in% assignmentOperators` and no modification is needed.  But if a user has written `result <- foo(nimOptim(...))`, we need to transform that to two lines: `Interm1 <- nimOptim(...)` and `result <- foo(Interm1)`.  So when size processing `nimOptim`, we can see that its caller is not an assignment operator and perform the `sizeInsertIntermediate` step, which creates the two lines described.  Summary: if the caller is already an assignment operator, no step is needed.

I saw that other case and figured you had just imitated it.  I already ran into that and fixed it on branch nimbleList-RCfun, so you could either fix it on your branch or wait to pick up the change later.